pygmt/clib/conversion.py

```diff
@@ -286,7 +291,12 @@ def strings_to_ctypes_array(strings: Sequence[str]) -> ctp.Array:
     >>> [s.decode() for s in ctypes_array]
     ['first', 'second', 'third']
     """
-    return (ctp.c_char_p * len(strings))(*[s.encode() for s in strings])
+    try:
+        bytes_string_list = [s.encode() for s in strings]
+    except AttributeError:  # 'pyarrow.StringScalar' object has no attribute 'encode'
+        # Convert pyarrow.StringArray to Python list first
+        bytes_string_list = [s.encode() for s in strings.to_pylist()]
+    return (ctp.c_char_p * len(strings))(*bytes_string_list)
```

Comment:
Key part to review is the changes on the `strings_to_ctypes_array` function here (see also #3137). I'm converting the `pyarrow.StringArray` -> `list` -> `ctypes`, but wondering if there's a more efficient way to convert pyarrow to ctypes.